spec/datetimeformat.html

```diff
@@ -1245,6 +1083,168 @@ <h1>Abstract Operations for DateTimeFormat Objects</h1>
+          1. Let _bestFormat_ be DateTimeStyleFormat(_dateStyle_, _timeStyle_, _styles_).
+        1. Else,
+          1. Let _needDefaults_ be *true*.
+          1. If _required_ is ~date~ or ~any~, then
+            1. For each property name _prop_ of « *"weekday"*, *"year"*, *"month"*, *"day"* », do
+              1. Let _value_ be _formatOptions_.[[&lt;_prop_&gt;]].
+              1. If _value_ is not *undefined*, set _needDefaults_ to *false*.
+          1. If _required_ is ~time~ or ~any~, then
+            1. For each property name _prop_ of « *"dayPeriod"*, *"hour"*, *"minute"*, *"second"*, *"fractionalSecondDigits"* », do
+              1. Let _value_ be _formatOptions_.[[&lt;_prop_&gt;]].
+              1. If _value_ is not *undefined*, set _needDefaults_ to *false*.
+          1. If _needDefaults_ is *true* and _defaults_ is either ~date~ or ~all~, then
+            1. For each property name _prop_ of « *"year"*, *"month"*, *"day"* », do
+              1. Set _formatOptions_.[[&lt;_prop_&gt;]] to *"numeric"*.
+          1. If _needDefaults_ is *true* and _defaults_ is either ~time~ or ~all~, then
+            1. For each property name _prop_ of « *"hour"*, *"minute"*, *"second"* », do
+              1. Set _formatOptions_.[[&lt;_prop_&gt;]] to *"numeric"*.
+          1. Let _formats_ be _resolvedLocaleData_.[[formats]].[[&lt;_resolvedCalendar_&gt;]].
+          1. If _formatMatcher_ is *"basic"*, then
+            1. Let _bestFormat_ be BasicFormatMatcher(_formatOptions_, _formats_).
+          1. Else,
+            1. Let _bestFormat_ be BestFitFormatMatcher(_formatOptions_, _formats_).
+        1. Set _dateTimeFormat_.[[DateTimeFormat]] to _bestFormat_.
+        1. If _bestFormat_ has a field [[hour]], then
+          1. Set _dateTimeFormat_.[[HourCycle]] to _hc_.
+        1. Return _dateTimeFormat_.
+      </emu-alg>
+    </emu-clause>
+
+    <emu-clause id="sec-formatoffsettimezoneidentifier" type="abstract operation">
+      <h1>
+        FormatOffsetTimeZoneIdentifier (
+          _offsetMinutes_: an integer,
+        ): a String
+      </h1>
+      <dl class="header">
+        <dt>description</dt>
+        <dd>
+          It formats a UTC offset, in minutes, into a UTC offset string formatted like ±HH:MM.
+        </dd>
+      </dl>
+      <emu-alg>
+        1. If _offsetMinutes_ ≥ 0, let _sign_ be the code unit 0x002B (PLUS SIGN); otherwise, let _sign_ be the code unit 0x002D (HYPHEN-MINUS).
+        1. Let _absoluteMinutes_ be abs(_offsetMinutes_).
+        1. Let _hours_ be floor(_absoluteMinutes_ / 60).
+        1. Let _minutes_ be _absoluteMinutes_ modulo 60.
+        1. Return the string-concatenation of _sign_, ToZeroPaddedDecimalString(_hours_, 2), the code unit 0x003A (COLON), and ToZeroPaddedDecimalString(_minutes_, 2).
+      </emu-alg>
+    </emu-clause>
+
     <emu-clause id="sec-date-time-style-format" type="abstract operation">
       <h1>
         DateTimeStyleFormat (
```

Comment:
Nit for a followup (if only to minimize review burden): we should also consider alphabetizing these AO lists, and moving/reorganizing where that would hinder comprehension.

Reply:
I'm provisionally skeptical about alphabetizing the AO lists, largely because the decision on when to break from alphabetical order for higher comprehension is difficult / might introduce inconsistencies that are themselves a source of confusion. I'd almost rather just have the rule be alphabetical order, full stop, no exceptions. Open to arguments otherwise, though -- this isn't a strong preference, and it's likely a somewhat uninformed one.

Reply:
To clarify, that's in line with my suggestion... the kind of moving/reorganization that I suggest, where necessary, relates to document structure (e.g., [FilterLocales](https://tc39.es/ecma402/#sec-filterlocales) and [ResolveLocale](https://tc39.es/ecma402/#sec-resolvelocale) could be alphabetically sorted as the only children of a "Locale Matching" subsection).

Reply:
ah, that makes a lot of sense! I've put it on my todo list